src/zenml/config/stack_component_resource_settings.py

```diff
@@ -0,0 +1,27 @@
+from typing import Set
+
+from zenml.config.base_settings import BaseSettings
+
+
+class StackComponentResourceSettings(BaseSettings):
```

Comment:
I am a bit confused here. Initially, I thought the implementation for the stack component resource settings would also inherit from the `ResourceSettings`. I am probably missing out on something here. Could you perhaps explain the relationship between the `ResourceSettings` and `StackComponentResourceSettings`?

Reply:
I wanted to make it inherit from that, but that would defeat the purpose because then they'd inherit even the attributes that are not possible to specify for this component.

Or do you think it's better to make them inherit from it, and the have a class var or something that defines which are ignored?

Reply:
@bcdurak Any opinion on this? It's either:
- We inherit from `ResourceSettings`, and somehow have a class var or similar that defines which resource settings are not available. In that case, we can even make it fail with some validator if one of these is passed using this explicit class?
- We don't inherit from it. I guess we save this class var, and also users would not get autocomplete for attributes that are not available.

Reply:
Some more points:
- Inheritance: If a new attribute gets added to resource settings, we need to add the "not supported flags" to all orchestrators/step operators.
- No inheritance: We need to redefine all resource settings attributes on all the subclasses that support them